### DISCUSSION TITLE
Screenshot contains warning "please select an object before switching to ocular view"

### DISCUSSION THREAD

**Discussion by sneakyimp** [Stellarium/stellarium#2364] [Q&A] [answered]
I downloaded Stellarium v0.21.3 from stellarium.org yesterday and installed it on my Mac. I then tried to take a screenshot of a certain sky region which I navigate to with a script (see below). Unfortunately, right in the middle of the screenshot, there is a warning in green text "**Please select an object before switching to ocular view**".

I am *not* trying to switch to ocular view, and wonder why this warning appears in my screenshot? I am just trying to get a *large* screenshot on my 4k monitor of a specific region of night sky without any lines or labels or warnings. I wonder if perhaps the MacOS quick key combination might also be assigned in Stellarium to something related to ocular view?  **How can I eliminate this warning from my screenshot?**

I would point out that the hot key combination for taking a full screen screenshot on MacOS is [Shift-Command-3](https://support.apple.com/en-us/HT201361). I wonder if perhaps this key combo might also be assigned in Stellarium?

NOTE that I was running this script to point Stellarium at a particular sky region on a particular date:
```
LandscapeMgr.setFlagLandscape(false);
LandscapeMgr.setFlagAtmosphere(false);
LandscapeMgr.setFlagFog(false);
LandscapeMgr.setFlagCardinalsPoints(true);
LandscapeMgr.setFlagAtmosphere(false),
StelSkyLayerMgr.setFlagShow(false);
MilkyWay.setFlagShow(true);
MilkyWay.setIntensity(0.5);
ZodiacalLight.setFlagShow(false);
GridLinesMgr.setFlagGridlines(false);
ConstellationMgr.setFlagLines(false);
ConstellationMgr.setFlagLabels(false);
SolarSystem.setFlagLabels(false);
StarMgr.setFlagLabels(false);
core.setObserverLocation(2.8867492, 50.8510759, 20);
core.moveToAltAzi(17, 275);
StelMovementMgr.zoomTo(38, 0.5);
// April 19, 1915
core.setDate("1915-04-19T20:45:00");
```

**Comment by gzotti** [comment, marked as answer]
Yes, this looks like a hotkey system conflict. 

- Screenshots should be made with the built-in function, by default assigned to Ctrl+S. (See F1/Help for keys on your Mac). 
- You can check or reassign keys from the F1/Help dialog.

**Reply by sneakyimp** [reply]
@gzotti *thank you* for your response. Where does Stellarium put the screenshots on MacOS? Also, I do have a 4k screen, I'm wondering if [this bug](https://github.com/Stellarium/stellarium/issues/102) is still a problem?

**Reply by gzotti** [reply]
Please see chapter 5 of the User Guide for directories. I don't know MacOS specifics.
The bug report from 2018 is marked as solved, so it should be just that.

**Comment by RamoneChook** [comment]
I use stellarium for mac, and when I try to take a screenshot (Shift-Command-3) it always works. With you there must be a hotkeys system, therefore try using Shift-Command-4, this will let you select the area you would like to screenshot.
So basically just select the whole screen with Shift-Command-4.
Unless there is a hotkeys system for Shift-Command-4, this will always work.
Keep in mind that there is another solution...
Go to system preferences and search up "Screenshot". This should take you to keyboard shortcuts where you may modify the keyboard shortcut for taking a screenshot. Example: Change Shift-Command-3 to Shift-Option-7.
This should always work.
Hopefully this helped. 🙂
